### ISSUE TITLE
Explore: Display properties not updating correctly

### ISSUE THREAD

**Issue by vereecw** [grafana/grafana#63180]
**What happened**:
A custom datasource backend plugin we're developing has the option to include display properties for a field such as the unit. This can be toggled with a switch in the query editor. When using this in a dashboard the resulting graph will include/exclude these display properties depending on if the switch is toggled or not. But when we try this in explore it will keep the display properties of the first successful query.

**What you expected to happen**:
Explore should also update the display properties correctly like in a dashboard.

**Screenshots and query results**:

<details>
  <summary>Query result with display properties</summary>

```json
    {
      "request": {
        "url": "api/ds/query",
        "method": "POST",
        "data": {
          "queries": [
            {
              "refId": "A",
              "datasource": {
                "type": "factry-historian-datasource",
                "uid": "PFgEj8A4k"
              },
              "tabIndex": 1,
              "selectedAssetPath": "a5b19cd0-971e-11ed-87e2-0242c0a86004",
              "selectedAssetProperties": [
                "temperature"
              ],
              "queryType": "MeasurementQuery",
              "query": {
                "Database": "7b6af020-971e-11ed-b21d-0242c0a86004",
                "Options": {
                  "GroupBy": [
                    "status"
                  ],
                  "Aggregation": {
                    "Name": "mean",
                    "Period": ""
                  },
                  "IncludeLastKnownPoint": false,
                  "UseEngineeringSpecs": true
                },
                "Measurements": [
                  "9464d1fe-971e-11ed-b21d-0242c0a86004"
                ]
              },
              "selectedMeasurement": "event calc",
              "key": "Q-e15be4b8-9041-4245-bca1-a118493e0161-0",
              "datasourceId": 1,
              "intervalMs": 2000,
              "maxDataPoints": 1863
            }
          ],
          "range": {
            "from": "2023-02-09T09:16:40.284Z",
            "to": "2023-02-09T10:16:40.284Z",
            "raw": {
              "from": "now-1h",
              "to": "now"
            }
          },
          "from": "1675934200284",
          "to": "1675937800284"
        },
        "hideFromInspector": false
      },
      "response": {
        "results": {
          "A": {
            "status": 500,
            "frames": [
              {
                "schema": {
                  "name": "event calc",
                  "refId": "A",
                  "fields": [
                    {
                      "name": "time",
                      "type": "time",
                      "typeInfo": {
                        "frame": "time.Time"
                      },
                      "labels": {}
                    },
                    {
                      "name": "event calc",
                      "type": "number",
                      "typeInfo": {
                        "frame": "float64",
                        "nullable": true
                      },
                      "labels": {
                        "status": "Good"
                      },
                      "config": {
                        "unit": "example unit",
                        "min": 0,
                        "max": 10,
                        "thresholds": {
                          "mode": "absolute",
                          "steps": [
                            {
                              "value": 0,
                              "color": "red"
                            },
                            {
                              "value": 3,
                              "color": "green"
                            },
                            {
                              "value": 7,
                              "color": "red"
                            }
                          ]
                        }
                      }
                    }
                  ]
                },
                "data": {
                  "values": [
                    [
                      1675934232000,
                      1675934292000,
                      1675934352000,
                      1675934412000,
                      1675934472000,
                      1675934532000,
                      1675934592000,
                      1675934652000,
                      1675934712000,
                      1675934772000,
                      1675934832000,
                      1675934892000,
                      1675934952000,
                      1675935012000,
                      1675935072000,
                      1675935132000,
                      1675935192000,
                      1675935252000,
                      1675935312000,
                      1675935372000,
                      1675935432000,
                      1675935492000,
                      1675935552000,
                      1675935612000,
                      1675935672000,
                      1675935732000,
                      1675935792000,
                      1675935852000,
                      1675935912000,
                      1675935972000,
                      1675936032000,
                      1675936092000,
                      1675936152000,
                      1675936212000,
                      1675936272000,
                      1675936332000,
                      1675936392000,
                      1675936452000,
                      1675936512000,
                      1675936572000,
                      1675936632000,
                      1675936692000,
                      1675936752000,
                      1675936812000,
                      1675936872000,
                      1675936932000,
                      1675936992000,
                      1675937052000,
                      1675937112000,
                      1675937172000,
                      1675937232000,
                      1675937292000,
                      1675937352000,
                      1675937412000,
                      1675937472000,
                      1675937532000,
                      1675937592000,
                      1675937652000,
                      1675937712000,
                      1675937772000
                    ],
                    [
                      6.174328386839722,
                      1.2580985878126125,
                      3.272514081816778,
                      5.841896358970258,
                      6.620752536487444,
                      2.553167321165432,
                      1.9472422358165464,
                      6.214035448633065,
                      0.11871759191552657,
                      9.388688289062037,
                      4.074707594362225,
                      3.93488260554558,
                      3.4666440485019665,
                      4.828007217384198,
                      0.9077632447596845,
                      2.4540592210062773,
                      4.620994537574854,
                      1.3843463042032422,
                      9.879060142735181,
                      3.5261078726009885,
                      2.0569159006742064,
                      8.814202270136727,
                      9.897810913530385,
                      6.906075631459824,
                      4.1333958937245345,
                      6.483176532827026,
                      1.9504100054801938,
                      8.767513539693196,
                      3.2670955954444776,
                      2.4113498015355272,
                      8.438147941624608,
                      5.974279268302469,
                      3.0761128246979434,
                      2.025714654696579,
                      0.022491268229470375,
                      6.428263647682243,
                      8.09422185731123,
                      9.446682403977203,
                      2.057912445009221,
                      5.545552006292617,
                      7.122438860469638,
                      2.8465144889160134,
                      2.377911470876745,
                      4.95653206388728,
                      4.4756391816834435,
                      9.408587770841658,
                      2.6944960601605694,
                      5.8078115945578945,
                      1.364465188821198,
                      4.832046937461517,
                      5.540463217169905,
                      8.028833117902947,
                      8.686111836554353,
                      3.6934655876731446,
                      0.37645229783252676,
                      9.779941943978777,
                      7.492975433353125,
                      0.04676750633384678,
                      6.327663829291229,
                      6.109530840674756
                    ]
                  ]
                }
              }
            ],
            "refId": "A"
          }
        }
      }
    }
```
</details>

![with-display-properties](https://user-images.githubusercontent.com/90846062/217787331-36962233-41e4-43ec-ab3b-b864b7ee9ec5.png)

<details>
  <summary>Query result without display properties</summary>

```json
{
  "request": {
    "url": "api/ds/query",
    "method": "POST",
    "data": {
      "queries": [
        {
          "refId": "A",
          "datasource": {
            "type": "factry-historian-datasource",
            "uid": "PFgEj8A4k"
          },
          "tabIndex": 1,
          "selectedAssetPath": "a5b19cd0-971e-11ed-87e2-0242c0a86004",
          "selectedAssetProperties": [
            "temperature"
          ],
          "queryType": "MeasurementQuery",
          "query": {
            "Database": "7b6af020-971e-11ed-b21d-0242c0a86004",
            "Options": {
              "GroupBy": [
                "status"
              ],
              "Aggregation": {
                "Name": "mean",
                "Period": ""
              },
              "IncludeLastKnownPoint": false,
              "UseEngineeringSpecs": false
            },
            "Measurements": [
              "9464d1fe-971e-11ed-b21d-0242c0a86004"
            ]
          },
          "selectedMeasurement": "event calc",
          "key": "Q-e15be4b8-9041-4245-bca1-a118493e0161-0",
          "datasourceId": 1,
          "intervalMs": 2000,
          "maxDataPoints": 1863
        }
      ],
      "range": {
        "from": "2023-02-09T09:24:01.750Z",
        "to": "2023-02-09T10:24:01.750Z",
        "raw": {
          "from": "now-1h",
          "to": "now"
        }
      },
      "from": "1675934641750",
      "to": "1675938241750"
    },
    "hideFromInspector": false
  },
  "response": {
    "results": {
      "A": {
        "status": 500,
        "frames": [
          {
            "schema": {
              "name": "event calc",
              "refId": "A",
              "fields": [
                {
                  "name": "time",
                  "type": "time",
                  "typeInfo": {
                    "frame": "time.Time"
                  },
                  "labels": {}
                },
                {
                  "name": "event calc",
                  "type": "number",
                  "typeInfo": {
                    "frame": "float64",
                    "nullable": true
                  },
                  "labels": {
                    "status": "Good"
                  },
                  "config": {
                    "unit": "none"
                  }
                }
              ]
            },
            "data": {
              "values": [
                [
                  1675934653000,
                  1675934713000,
                  1675934773000,
                  1675934833000,
                  1675934893000,
                  1675934953000,
                  1675935013000,
                  1675935073000,
                  1675935133000,
                  1675935193000,
                  1675935253000,
                  1675935313000,
                  1675935373000,
                  1675935433000,
                  1675935493000,
                  1675935553000,
                  1675935613000,
                  1675935673000,
                  1675935733000,
                  1675935793000,
                  1675935853000,
                  1675935913000,
                  1675935973000,
                  1675936033000,
                  1675936093000,
                  1675936153000,
                  1675936213000,
                  1675936273000,
                  1675936333000,
                  1675936393000,
                  1675936453000,
                  1675936513000,
                  1675936573000,
                  1675936633000,
                  1675936693000,
                  1675936753000,
                  1675936813000,
                  1675936873000,
                  1675936933000,
                  1675936993000,
                  1675937053000,
                  1675937113000,
                  1675937173000,
                  1675937233000,
                  1675937293000,
                  1675937353000,
                  1675937413000,
                  1675937473000,
                  1675937533000,
                  1675937593000,
                  1675937653000,
                  1675937713000,
                  1675937773000,
                  1675937833000,
                  1675937893000,
                  1675937953000,
                  1675938013000,
                  1675938073000,
                  1675938133000,
                  1675938193000
                ],
                [
                  6.214035448633065,
                  0.11871759191552657,
                  9.388688289062037,
                  4.074707594362225,
                  3.93488260554558,
                  3.4666440485019665,
                  4.828007217384198,
                  0.9077632447596845,
                  2.4540592210062773,
                  4.620994537574854,
                  1.3843463042032422,
                  9.879060142735181,
                  3.5261078726009885,
                  2.0569159006742064,
                  8.814202270136727,
                  9.897810913530385,
                  6.906075631459824,
                  4.1333958937245345,
                  6.483176532827026,
                  1.9504100054801938,
                  8.767513539693196,
                  3.2670955954444776,
                  2.4113498015355272,
                  8.438147941624608,
                  5.974279268302469,
                  3.0761128246979434,
                  2.025714654696579,
                  0.022491268229470375,
                  6.428263647682243,
                  8.09422185731123,
                  9.446682403977203,
                  2.057912445009221,
                  5.545552006292617,
                  7.122438860469638,
                  2.8465144889160134,
                  2.377911470876745,
                  4.95653206388728,
                  4.4756391816834435,
                  9.408587770841658,
                  2.6944960601605694,
                  5.8078115945578945,
                  1.364465188821198,
                  4.832046937461517,
                  5.540463217169905,
                  8.028833117902947,
                  8.686111836554353,
                  3.6934655876731446,
                  0.37645229783252676,
                  9.779941943978777,
                  7.492975433353125,
                  0.04676750633384678,
                  6.327663829291229,
                  6.109530840674756,
                  4.110732328417548,
                  7.504466461701048,
                  3.703693466846016,
                  8.024262339331441,
                  5.801528030296966,
                  7.157833173613069,
                  7.491048234648435
                ]
              ]
            }
          }
        ],
        "refId": "A"
      }
    }
  }
}
```
</details>

![without-display-properties](https://user-images.githubusercontent.com/90846062/217787407-b984e7f5-a493-4979-a0f7-9ddbd56e0c93.png)

**Environment**:
- Grafana version: 9.3.6

**Comment by usmangt** [comment]
Hi @vereecw,

Thanks for opening this issue.

Since this is a custom datasource plugin. Is it possible to access it (even have dummy data to test it out)?

Also, can you show (share a screenshot) how it looks like while it works correctly in dashboard?

Anything in the backend logs OR maybe in the browser console when viewing it in the Explore module?

**Comment by gelicia** [comment]
Hi @vereecw - this is a bug that was fixed in #62473 and will be released in 9.4.0. Unfortunately it is not released yet, but it is planned on being released at the end of the month. Sorry for the mixed news. Feel free to re-open if you still experience this problem after that.

**Comment by gelicia** [comment]
**Replication steps** (for future reference)

Tested against play.grafana.org (running 9.3.6 at the time) and local (running against `main`)

<details>
<summary>Frame 1: Time ends in 3000, value is 6.21. Config.unit: none</summary>

```
[
          {
            "schema": {
              "name": "event calc",
              "refId": "A",
              "fields": [
                {
                  "name": "time",
                  "type": "time",
                  "typeInfo": {
                    "frame": "time.Time"
                  },
                  "labels": {}
                },
                {
                  "name": "event calc",
                  "type": "number",
                  "typeInfo": {
                    "frame": "float64",
                    "nullable": true
                  },
                  "labels": {
                    "status": "Good"
                  },
                  "config": {
                    "unit": "none"
                  }
                }
              ]
            },
            "data": {
              "values": [
                [
                  1675934653000
                ],
                [
                  6.214035448633065
                ]
              ]
            }
          }
        ]
```  
</details>


<details>
<summary>Frame 2: Time ends in 2000, value is 6.17. Config.units: "example unit"</summary>

```
[
              {
                "schema": {
                  "name": "event calc",
                  "refId": "A",
                  "fields": [
                    {
                      "name": "time",
                      "type": "time",
                      "typeInfo": {
                        "frame": "time.Time"
                      },
                      "labels": {}
                    },
                    {
                      "name": "event calc",
                      "type": "number",
                      "typeInfo": {
                        "frame": "float64",
                        "nullable": true
                      },
                      "labels": {
                        "status": "Good"
                      },
                      "config": {
                        "unit": "example unit"
                      }
                    }
                  ]
                },
                "data": {
                  "values": [
                    [
                      1675934232000
                    ],
                    [
                      6.174328386839722
                    ]
                  ]
                }
              }
            ]
```  
</details>

Using the "testdata" datasource with the "Raw Frames" scenario, copy one frame, then replace it with the other frame. In 9.3.6, the config.unit of the first frame will persist even when replaced with another frame that has a different unit. As of 9.4, it replaces the unit as well.
